core/templates/sort_array.h

```diff
@@ -60,7 +60,78 @@ class SortArray {
 public:
 	Comparator compare;
 
-	inline const T &median_of_3(const T &a, const T &b, const T &c) const {
+	/**
```

Comment:
We don't normally add doc comments like this, so I'd say to remove them, especially without having them generally

Reply:
Do you mean I should use `//` instead of `/**` or that I should remove the comments altogether ?

If it's the later I don't mind but IMO `nth_element` at least needs some kind of explanation because its usage is really not obvious (unless you know STL has a similarly named function).
I think it would also be beneficial to keep the explanation about the sorting algorithm not being stable.

Reply:
Remove them entirely, you can put some details there but not a full doc comment explaining it in detail

Reply:
Ok, will do.

Reply:
Fixed all the review comments.
As for the method docs, I only left what I thought was strictly necessary.